charts/cla/templates/deployment.yaml

```diff
@@ -0,0 +1,82 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "cla.fullname" . }}
+  labels:
+    {{- include "cla.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "cla.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "cla.selectorLabels" . | nindent 8 }}
+    spec:
+    {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      serviceAccountName: {{ include "cla.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+          - name: HOST
+            value: {{ .Values.claAssistant.host }}
+          - name: PORT
+            value: {{ .Values.claAssistant.port | default 80 }}
+          - name: HOST_PORT
+            value: {{ .Values.claAssistant.hostPort | default 8080 }}
+          - name: PROTOCOL
+            value: {{ .Values.claAssistant.protocol | default "http" }}
+          - name: GITHUB_CLIENT
+            value: {{ .Values.githubClient | default "CLIENT" }}
+          - name: GITHUB_SECRET
+            valueFrom:
+              secretKeyRef:
+                name: github-credentials
+                key: secret
+          - name: GITHUB_USER
+            value: "Hello from the environment"
+          - name: GITHUB_PASS
+            valueFrom:
+              secretKeyRef:
+                name: github-credentials
+                key: pass
+          - name: GITHUB_TOKEN
+            valueFrom:
+              secretKeyRef:
+                name: github-credentials
+                key: token
+          - name: GITHUB_ADMIN_USERS
+            value: "Such a sweet sorrow"
+          - name: MONGODB
+            valueFrom:
+              secretKeyRef:
+                name: mongodb credentials
```

Comment:
We should use a template partial here too. Also, AFAIK a name with a space in it won't work.